src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -453,13 +556,27 @@ DatabasePtr DatabaseCatalog::detachDatabase(ContextPtr local_context, const Stri
     DatabasePtr db;
     {
         std::lock_guard lock{databases_mutex};
-        assertDatabaseExistsUnlocked(database_name);
         db = databases.find(database_name)->second;
```

Comment:
this will not work - what if `database_name` doesn't exist in `databases`? it will dereference 'end' iterator...